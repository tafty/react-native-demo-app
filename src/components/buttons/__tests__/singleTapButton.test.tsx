import 'react-native'
import React from 'react'
import { Text } from 'react-native'
import { fireEvent, render } from 'react-native-testing-library'

import SingleTapButton from '../singleTapButton'

describe('SingleTapButton', () => {
  it('should render', () => {
    const testId = 'test-id'
    const children = <Text testID={testId}>Test</Text>
    const onPress = jest.fn()

    const { getByTestId, toJSON } = render(
      <SingleTapButton children={children} onPress={onPress} />,
    )

    fireEvent.press(getByTestId(testId))
    expect(onPress).toBeCalled()

    expect(getByTestId(testId).props.children).toBe('Test')

    expect(toJSON()).toMatchSnapshot()
  })
})
