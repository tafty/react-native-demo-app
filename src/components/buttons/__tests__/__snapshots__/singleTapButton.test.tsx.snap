// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleTapButton should render 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    testID="test-id"
  >
    Test
  </Text>
</View>
`;
